include/lsst/sphgeom/codec.h

```diff
@@ -80,6 +80,44 @@ inline double decodeDouble(uint8_t const * buffer) {
 #endif
 }
 
+/// `encodeU64` appends an uint64 in little-endian byte order
+/// to the end of buffer.
+inline void encodeU64(std::uint64_t item, std::vector<uint8_t> & buffer) {
+#ifdef OPTIMIZED_LITTLE_ENDIAN
+    auto ptr = reinterpret_cast<uint8_t const *>(&item);
+    buffer.insert(buffer.end(), ptr, ptr + 8);
+#else
+    union { uint64_t u; double d; };
+    d = item;
+    buffer.push_back(static_cast<uint8_t>(u));
```

Comment:
You might consider a `resize()` here and then just using array subscripting assignments, to avoid the redundant bounds checking in eight successive `push_back()` calls.  OTOH, there are less than ten of them, so it's probably a <shrug> :-)

Reply:
I'm a little worried it could be an accidental pessimization, because the pattern is that lots of functions like this are going to be pushing into the same `buffer` vector, and I don't actually know if `resize` is going to (when the capacity isn't big enough) allocate _just_ the amount of extra space needed or do the kind of "allocate more than you did last time" thing that keeps `push_back` amortized constant-time.  Do you happen to know?

Reply:
Good point.  I think you could call `capacity()` before `resize()` to avoid the undesired behavior, but I have also just realized the the `resize()` is going to have to do a default init on all eight slots, so that will have a cost as well.   

In any case, for eight items per batch I think it's hard to argue convincingly that there would be much of a win or loss here in the end.